### PR TITLE
Version 1.2 full review

### PR DESCRIPTION
**Pull request by ivang-coder**
Can you do a full review? Would appreciate your advice. Cheers.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,66 @@
 # rexifier
-rEXIFier is a Shell-written EXIF-based media file processor with sort, dedup and other features
+
+rEXIFier is a Shell-written EXIF-based media file processor with sort, dedup and other features.
+
+ This script is used to sort media files (jpg, jpeg, mp4, mpg and mov ). 
+ It uses EXIFTOOL to extract "SubSecCreateDate", "CreateDate" and 
+ "Model" metadata. In case EXIF attempts fail and if the appropriate flad is enabled, then less reliable file attributes get
```
Comment (annag-coder) [inline README.md:7]: `flad` ?

File: rEXIFier.sh [hunk 2]
```diff
@@ -755,23 +705,63 @@ for WIPSortedFileAbsolutePath in ${WIPSortedFileAbsolutePaths[@]}; do
...
+                # Setting the value if the filename does not contain a duplicate index
+                if [[ LastSortedMatchedFileDuplicateIndex == "" ]] ; then
+                  LastSortedMatchedFileDuplicateIndex="00"
+                fi
+                # Incrementing the duplicate index so that we can form a new File Name
+                printf -v IncrementedLastSortedMatchedFileDuplicateIndex %02d "$((10#$LastSortedMatchedFileDuplicateIndex + 1))"
+                # Passing the incremented duplicate index to IncrementedDuplicateIndex
+                IncrementedDuplicateIndex=${IncrementedLastSortedMatchedFileDuplicateIndex}
+                # Re-forming the File Name with the incremented duplicate index
+                FormatedFileName=${FormatedFileNameNoExtension}-${IncrementedDuplicateIndex}.${NormalisedFileExtension}
```
Comment (annag-coder) [inline rEXIFier.sh]: it feels like there is a lot of work just to find out the increment number, would be simpler just to add a random number at the end

File: rEXIFier.sh [hunk 3]
```diff
@@ -783,10 +773,51 @@ for WIPSortedFileAbsolutePath in ${WIPSortedFileAbsolutePaths[@]}; do
...
+                  # Setting the value if the filename does not contain a duplicate index
+                  if [[ LastSortedMatchedFileDuplicateIndex == "" ]] ; then
+                    LastSortedMatchedFileDuplicateIndex="00"
+                  fi
+                  # Incrementing the duplicate index so that we can form a new File Name
+                  printf -v IncrementedLastSortedMatchedFileDuplicateIndex %02d "$((10#$LastSortedMatchedFileDuplicateIndex + 1))"
+                  # Passing the incremented duplicate index to IncrementedDuplicateIndex
+                  IncrementedDuplicateIndex=${IncrementedLastSortedMatchedFileDuplicateIndex}
+                  # Re-forming the File Name with the incremented duplicate index
+                  FormatedFileName=${FormatedFileNameNoExtension}-${IncrementedDuplicateIndex}.${NormalisedFileExtension}
```
Comment (annag-coder) [inline rEXIFier.sh:774]: this whole block of code does the same thing as above, would be better to move to a reusable function

**Comment by ivang-coder** [comment]
The pull request is obsolete now. Closing it.

**Pull request closed without merging**